ArduPlane/reverse_thrust.cpp

```diff
@@ -61,6 +61,10 @@ bool Plane::allow_reverse_thrust(void) const
         allow |= (g.use_reverse_thrust & USE_REVERSE_THRUST_AUTO_WAYPOINT) &&
                     (nav_cmd == MAV_CMD_NAV_WAYPOINT ||
                      nav_cmd == MAV_CMD_NAV_SPLINE_WAYPOINT);
+
+        // we are on a landing pattern
+        allow |= (g.use_reverse_thrust & USE_REVERSE_THRUST_AUTO_LANDING_PATTERN) &&
```

Comment:
How does this ensure you are actually in auto and trying to run a landing? Seems like if you were doing a landing, then went FBWB or something like it would still be allowing reverse thrust?

Reply:
true, a mode check is needed

Reply:
actually, this whole thing is in a switch for mode auto

Reply:
Ah, so it is, sorry!